pkg/kubectl/cmd/run.go

```diff
@@ -461,7 +461,7 @@ func (o *RunOptions) removeCreatedObjects(f cmdutil.Factory, createdObjects []*R
 			Internal(legacyscheme.Scheme).
 			ContinueOnError().
 			NamespaceParam(namespace).DefaultNamespace().
-			ResourceNames(obj.Mapping.Resource, name).
+			ResourceNames(obj.Mapping.Resource.Resource+"."+obj.Mapping.Resource.Group, name).
```

Comment:
previously, if `obj.Mapping.Resource` was empty, this would fall back to expect `name` to be `<resourceType>/<resourceName>`. Is it possible for obj.Mapping.Resource.Resource to be empty here and to end up passing `"."` to ResourceNames?

Reply:
You don't get empty resources back in mappings.